packages/react-refresh/src/__tests__/__snapshots__/ReactFreshBabelPlugin-test.js.snap

```diff
@@ -110,21 +110,21 @@ exports[`ReactFreshBabelPlugin generates signatures for function expressions cal
 var _s = $RefreshSig$(),
     _s2 = $RefreshSig$();
 
-export const A = _s(React.memo(_c2 = React.forwardRef(_c = (props, ref) => {
+export const A = _s(React.memo(_c2 = _s(React.forwardRef(_c = _s((props, ref) => {
   _s();
 
   const [foo, setFoo] = useState(0);
   React.useEffect(() => {});
   return <h1 ref={ref}>{foo}</h1>;
-})), "useState{[foo, setFoo](0)}\\nuseEffect{}");
+}, "useState{[foo, setFoo](0)}\\nuseEffect{}")), "useState{[foo, setFoo](0)}\\nuseEffect{}")), "useState{[foo, setFoo](0)}\\nuseEffect{}");
```

Comment:
This adds a bunch of duplication to the output. But the real one is hashed. Also it's only for HOCs which are out of favor. So meh.

Reply:
> Also it's only for HOCs which are out of favor. So meh.

Then again, isn't this change itself to support an even more uncommon use case (a HOC like pattern that directly calls a component as a function)?

Reply:
Well, HOCs are not _that_ uncommon.

In the original case, we have a roughly 7% usage library which downright causes a crash every time a component is edited, in default setups with CRA/Next/RN. Pretty disruptive and they don't have a fix without giving up on the whole approach.

I'm just saying that the added output duplication does not make the remaining cases worse. It just slightly bloats the DEV-only output.